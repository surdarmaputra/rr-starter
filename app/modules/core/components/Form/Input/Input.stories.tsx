import type { Meta, StoryObj } from '@storybook/react';

import { Input, type InputProps } from './Input';

type Story = StoryObj<typeof Input>;

const meta: Meta = {
  component: Input,
  title: 'Core/Components/Form/Input',
  argTypes: {
    disabled: {
      control: { type: 'boolean' },
    },
    placeholder: {
      control: { type: 'text' },
    },
    value: {
      control: { type: 'text' },
    },
  },
};

export default meta;

export const Default: Story = {
  args: {
    placeholder: 'Input some text...',
  },
  render: (args: InputProps) => <Input {...args} />,
};
