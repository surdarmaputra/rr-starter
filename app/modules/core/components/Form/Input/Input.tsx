import * as React from 'react';

import { cn } from '~/libs/shadcn/utils';

export type InputProps = React.ComponentProps<'input'>;

const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ className, type, ...props }, ref) => {
    return (
      <input
        className={cn(
          'flex h-10 w-full rounded-lg border border-divider bg-white px-3 py-2 text-base ring-offset-white',
          'file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-letter-title',
          'placeholder:text-letter-caption focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-primary focus-visible:ring-offset-1',
          'disabled:cursor-not-allowed disabled:opacity-50',
          'md:text-sm',
          className,
        )}
        ref={ref}
        type={type}
        {...props}
      />
    );
  },
);
Input.displayName = 'Input';

export { Input };
