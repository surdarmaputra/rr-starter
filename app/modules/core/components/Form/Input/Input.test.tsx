import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, test, vi } from 'vitest';

import { Input, type InputProps } from './Input';

describe('Input', () => {
  function renderComponent(props?: InputProps) {
    return render(<Input data-testid="Input-testid" {...props} />);
  }

  test('renders with data-testid', async () => {
    renderComponent();
    await screen.findByTestId('Input-testid');
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  test('handles input change', async () => {
    const value = 'anyValue';
    const handleChange = vi.fn();
    renderComponent({
      onChange: handleChange,
      placeholder: 'Sample input',
    });

    const input = await screen.findByPlaceholderText(/Sample input/);
    fireEvent.change(input, { target: { value } });

    expect(input).toHaveValue(value);
    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(handleChange).toHaveBeenCalledWith(
      expect.objectContaining({
        target: expect.objectContaining({
          value,
        }),
      }),
    );
  });
});
