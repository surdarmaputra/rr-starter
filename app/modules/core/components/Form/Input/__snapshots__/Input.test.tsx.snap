// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > renders snapshot 1`] = `
<div>
  <input
    class="flex h-10 w-full rounded-lg border border-divider bg-white px-3 py-2 text-base ring-offset-white file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-letter-title placeholder:text-letter-caption focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-primary focus-visible:ring-offset-1 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    data-testid="Input-testid"
  />
</div>
`;
